plugin/internal/command.go

```diff
@@ -0,0 +1,148 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/puppetlabs/wash/activity"
+)
+
+// Command is a wrapper to exec.Cmd. It handles context-cancellation cleanup
+// and defines a String() method to make logging easier.
+type Command struct {
+	*exec.Cmd
```

Comment:
It may be more confusing than helpful to make this wrap `exec.Cmd` rather than having it as a named property. You override some of the core methods, so not sure others would behave correctly.

Reply:
According to https://travix.io/type-embedding-in-go-ba40dd4264df, Go's type embedding makes sure that embedded structs don't have access to anything in the embedding struct. Thus, any of `exec.Cmd`'s core methods should call only `exec.Cmd`'s core methods, meaning this embedding should be OK and should not affect anything in `exec.Cmd`.

Reply:
My point was if someone ran `Output` on our Command, it wouldn't use our Start/Wait methods and so probably wouldn't work as expected. So is it worth letting `exec.Cmd`'s `Output` method be exposed?

Reply:
Ohh I see. Sorry about misunderstanding what you meant.

Their implementations are pretty short, so may as well expose them in case they're ever needed.

Reply:
Hmm, maybe not. `Output` is a bit complicated because they use an internal io.Writer. We could copy it word for word, but I don't see a good reason to do that yet.

Guess I'll go with embedding a property, even if it means re-defining some exported fields as setters. Makes things more secure.